From d23b8da96c8c9629d14d7ea522708d7cee0c8786 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Tue, 4 Oct 2022 19:46:32 +0200
Subject: [PATCH] platform/wayland: add libdrm dependency when weston is
 enabled

Otherwise the build complains about drm_fourcc.h header missing.

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 platform/wayland/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/platform/wayland/CMakeLists.txt b/platform/wayland/CMakeLists.txt
index d926c30..a8ec652 100644
--- a/platform/wayland/CMakeLists.txt
+++ b/platform/wayland/CMakeLists.txt
@@ -56,6 +56,9 @@ if (COG_WESTON_DIRECT_DISPLAY)
         REQUIRED
     )
 
+    pkg_check_modules(LibDRM IMPORTED_TARGET REQUIRED libdrm>=2.4.71)
+    target_link_libraries(cogplatform-wl PRIVATE PkgConfig::LibDRM)
+
     pkg_search_module(LIBWESTON REQUIRED libweston-11-protocols libweston-10-protocols libweston-9-protocols libweston-8-protocols)
     pkg_get_variable(LIBWESTON_PKG_DATA_DIR ${LIBWESTON_MODULE_NAME} "pkgdatadir")
     add_wayland_protocol(cogplatform-wl CLIENT "${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml")
-- 
2.35.1

